server-common/src/main/java/org/apache/kafka/server/common/Features.java

```diff
@@ -53,10 +53,21 @@ public enum Features {
     private final String name;
     private final FeatureVersion[] featureVersions;
 
+    // The latest production version of the feature, owned and updated by the feature owner
+    // in the respective feature definition. The value should not be smaller than the default
+    // value calculated with {@link #defaultValue(MetadataVersion)}.
+    private final FeatureVersion latestProduction;
+
     Features(String name,
-             FeatureVersion[] featureVersions) {
+             FeatureVersion[] featureVersions,
+             FeatureVersion latestProduction) {
         this.name = name;
         this.featureVersions = featureVersions;
+        this.latestProduction = latestProduction;
+
+        if (defaultValue(MetadataVersion.LATEST_PRODUCTION) > latestProduction.featureLevel()) {
```

Comment:
Do we need to further validate the following?
1. The dependencies of the latestProduction is in production.
2. The dependencies of the default feature is in production and have a level <= the default level of the dependency.

Reply:
Added a validation for 
- The latest production value >= the default value.
- The dependencies of the latest production value <= their latest production values.
- The dependencies of the default value <= their default values.

> The dependencies of the default feature is in production

This is guaranteed when checking `The latest production value >= the default value`